components/locale_core/src/extensions/unicode/subdivision.rs

```diff
@@ -61,17 +61,64 @@ impl_tinystr_subtag!(
 /// # Examples
 ///
 /// ```
-/// use icu::locale::{
-///     extensions::unicode::{subdivision_suffix, SubdivisionId},
-///     subtags::region,
-/// };
+/// use icu::locale::locale;
+/// use icu::locale::extensions::unicode::key;
+/// use icu::locale::preferences::extensions::unicode::keywords::RegionOverride;
+/// use writeable::assert_writeable_eq;
 ///
-/// let ss = subdivision_suffix!("zzzz");
-/// let region = region!("gb");
+/// // American English with British user preferences
+/// let locale = locale!("en-US-u-rg-gbzzzz");
 ///
-/// let si = SubdivisionId::new(region, ss);
+/// let normal_region = locale.id.region.unwrap();
+/// let rg_extension_value = locale.extensions.unicode.keywords.get(&key!("rg")).unwrap();
+/// let subdivision = RegionOverride::try_from(rg_extension_value.clone()).unwrap();
```

Comment:
keep the names aligned - `rg` is `RegionOverride`, `sd` is `RegionalSubdivision`

```suggestion
/// let rg_extension_value = locale.extensions.unicode.keywords.get(&key!("rg")).unwrap();
/// let region_override = RegionOverride::try_from(rg_extension_value.clone()).unwrap();
```